libslide/src/bignum/add.rs

```diff
@@ -0,0 +1,134 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::Bignum;
+use std::mem;
+use std::ops;
+
+impl ops::Add for Bignum {
+    type Output = Bignum;
+
+    fn add(self, rhs: Bignum) -> Bignum {
+        let is_neg: bool = rhs.is_neg && self.is_neg;
+
+        // this assumes the signs are both positive for now
+        let mut carry: u8 = 0;
+        let mut lhs_size: usize = self.dec.len();
+        let mut rhs_size: usize = rhs.dec.len();
+        let mut lhs_vec: Vec<u8>;
+        let mut rhs_vec: Vec<u8>;
+        let mut res: u8;
+
+        // 1. Handle Decimal
+
+        let mut res_dec: Vec<u8>;
+
+        // make lhs smaller vector
+        if lhs_size > rhs_size {
```

Comment:
I think it might be easier to understand if you compare self.dec and rhs.dec sizes here, then set lhs_size and rhs_size inside the conditional.

Reply:
Yes, it would, but we would loose performance because we would need to calculate self.dec.len() twice. One here and once in the swap